azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/extensions/AzureComputeImageExtension.java

```diff
@@ -69,29 +73,28 @@
       this.userExecutor = userExecutor;
       this.group = azureComputeConstants.azureResourceGroup();
       this.imageReferenceToImage = imageReferenceToImage;
-      this.api = api;
       this.imageAvailablePredicate = imageAvailablePredicate;
       this.nodeSuspendedPredicate = nodeSuspendedPredicate;
       this.azureComputeConstants = azureComputeConstants;
+      this.api = api;
    }
 
    @Override
    public ImageTemplate buildImageTemplateFromNode(String name, String id) {
-      String imageName = name.toLowerCase();
-      return new ImageTemplateBuilder.CloneImageTemplateBuilder().nodeId(id).name(imageName).build();
+      name = name.toLowerCase();
```

Comment:
We should probably use local variables and assume immutable parameters. Don't modify references to an input variable.